### PR TITLE
Feature/col 194

### PR DESCRIPTION
**Pull request by matt-williams8**

### CODE REVIEW

File: projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts [hunk 1]
```diff
@@ -79,22 +84,33 @@ export class MerchandisingCarouselComponent {
     return metadata;
   }
 
-  private mapMerchandisingProductsItems(
+  private mapMerchandisingProductsToCarouselItems(
```
Comment (bgambocjaviniar) [inline projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts]: nice

File: projects/cds/src/merchandising/model/merchandising-products.model.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { Product } from '@spartacus/core';
 
+export interface MerchandisingProduct extends Product {
+  metadata?: Map<string, string>;
+}
+
 export interface MerchandisingProducts {
   metadata?: Map<string, string>;
```
Comment (bgambocjaviniar) [inline projects/cds/src/merchandising/model/merchandising-products.model.ts]: just for my understanding, 

there's a global metadata for 'products'
and a lower level metadata for a single 'product'

File: projects/cds/src/merchandising/model/strategy-result.model.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export interface MerchandisingProduct {
+export interface StrategyProduct {
```
Comment (bgambocjaviniar) [inline projects/cds/src/merchandising/model/strategy-result.model.ts]: Another thing to understand for me only,

what is the difference between merchandisingProduct and StrategyProduct

File: projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.html [hunk 4]
```diff
@@ -16,6 +16,12 @@
 </ng-container>
 
 <ng-template #carouselItem let-item="item">
+  <div
+    id="data-cx-merchandising-product-{{ item.code }}"
```
Comment (bgambocjaviniar) [inline projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.html]: is this used for something?

File: projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts [hunk 5]
```diff
@@ -79,22 +84,33 @@ export class MerchandisingCarouselComponent {
...
+  ): Observable<MerchandisingProduct>[] {
+    return merchandisingProducts && merchandisingProducts.products
+      ? merchandisingProducts.products.map((product, index) => {
+          product.metadata = this.getCarouselItemMetadata(product, index + 1);
+          return of(product);
+        })
+      : [EMPTY];
+  }
+
+  getCarouselItemMetadata(
```
Comment (znikola) [inline projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts]: Is this method used in the template? If not, please make it `private`.

File: projects/cds/src/merchandising/model/merchandising-products.model.ts (hunk 2 above)
Comment (matt-williams8) [reply projects/cds/src/merchandising/model/merchandising-products.model.ts]: That's right

File: projects/cds/src/merchandising/model/strategy-result.model.ts (hunk 3 above)
Comment (matt-williams8) [reply projects/cds/src/merchandising/model/strategy-result.model.ts]: A `StrategyProduct` is the model of the product as it is returned from our strategy service, then for ease of rendering the carousel we want to use the core spartacus `Product` model (as that is what the existing `cx-carousel` accepts as its item model). As we then need the product level metadata the easiest option seemed to be to extend the core `Product` model and add the metadata to it, that extension is the `MerchandisingProduct`

File: projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.html (hunk 4 above)
Comment (matt-williams8) [reply projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.html]: Yea, this is part of the metadata for profile-tag to scrape. Do you prefer that ids are not used?

File: projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts (hunk 5 above)
Comment (matt-williams8) [reply projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts]: Nope, I tried but I got "super responsiveness" again 😉 , will make it private

Comment (znikola) [reply projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts]: Haha, _reactive_ ftw :)

File: projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.html (hunk 4 above)
Comment (bgambocjaviniar) [reply projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.html]: it is fine :)

**Pull request merged**